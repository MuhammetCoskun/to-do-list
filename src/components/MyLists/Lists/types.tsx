import { ToDoListItemType } from "../../ShowList/ToDoList/types";

export type ListType = {
  id: string;
  title: string;
  todos: ToDoListItemType[];
};

export type ListsProps = {
  lists: ListType[];
  onChooseList: (arg: string) => void;
};
