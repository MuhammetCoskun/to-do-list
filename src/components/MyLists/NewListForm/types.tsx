export type NewListFormProps = {
  onAddList: (listName: string) => void;
};
