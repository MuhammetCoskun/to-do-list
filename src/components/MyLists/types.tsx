import { ListType } from "./Lists/types";

export type MyListsProps = {
  onChooseList: (arg: string) => void;
  onAddList: (title: string) => void;
  lists: ListType[];
};
