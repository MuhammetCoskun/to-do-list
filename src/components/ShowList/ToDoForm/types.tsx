export interface ToDoFormProps {
  onAddTodo: (todo: string) => void;
  onRemoveTodo: () => void;
  onRemoveList: () => void;
}
