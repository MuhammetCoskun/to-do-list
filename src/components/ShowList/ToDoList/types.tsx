import { ListType } from "../../MyLists/Lists/types";

export type ToDoListItemType = {
  id: string;
  text: string;
  createdAt: string;
  isDone: boolean;
  doneAt?: string;
};

export type ToDoListItemProps = {
  id: string;
  key?: string;
  text: string;
  onRemove: (id: string) => void;
};

export type ToDoListProps = {
  list: ListType;
  onRemove: (id: string) => void;
};
