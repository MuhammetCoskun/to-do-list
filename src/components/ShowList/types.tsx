import { ListType } from "../MyLists/Lists/types";

export type ShowListProps = {
  listToShow: ListType;
  onAddTodo: (text: string) => void;
  onRemoveTodo: (id: string) => void;
};
