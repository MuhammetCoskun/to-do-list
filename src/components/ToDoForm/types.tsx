export interface TodoFormProps {
  onAddTodo: (todo: string) => void;
}
