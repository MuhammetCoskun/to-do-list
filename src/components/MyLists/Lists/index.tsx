import React from "react";
import { ListsProps } from "./types";

const Lists: React.FC<ListsProps> = ({ lists, onChooseList }) => {
  return (
    <ul className="mylists-list">
      {lists.map((list) => (
        <li
          key={list.id}
          onClick={() => onChooseList(list.id)}
          className={"list-name active-list"}
        >
          {list.title}
        </li>
      ))}
    </ul>
  );
};

export default Lists;
