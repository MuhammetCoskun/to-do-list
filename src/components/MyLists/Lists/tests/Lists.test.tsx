import React from "react";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Lists from "../Lists";
import { ListType, ListsProps } from "../types";

const mockOnChooseList = jest.fn();
const mockList: ListType[] = [{ id: "1", title: "Test", todos: [] }];
const properties: ListsProps = {
  lists: mockList,
  onChooseList: mockOnChooseList,
};

const customRender = (props: ListsProps) => {
  render(<Lists {...props} />);
};
describe("Lists component", () => {
  test("should render correctly", () => {
    const { asFragment } = render(
      <Lists lists={mockList} onChooseList={mockOnChooseList} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
  test("should call onChooseList callback when list item is clicked", () => {
    customRender({ ...properties });
    const listElement = screen.getByTestId("1");
    userEvent.click(listElement);
    expect(mockOnChooseList).toBeCalledWith("1");
  });
});
