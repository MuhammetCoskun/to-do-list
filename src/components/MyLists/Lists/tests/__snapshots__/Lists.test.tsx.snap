// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists component should render correctly 1`] = `
<DocumentFragment>
  <ul
    class="mylists-list"
  >
    <li
      class="list-name active-list"
      data-testid="1"
    >
      Test
    </li>
  </ul>
</DocumentFragment>
`;
